### PR TITLE
Added optional 'platform name' to 'update' parameter

### PR DESCRIPTION
**Pull request by simongranger**

### CODE REVIEW

File: yi_appium_caps_util.gemspec [hunk 1]
```diff
@@ -1,8 +1,8 @@
 Gem::Specification.new do |s|
   s.name        = 'yi_appium_caps_util'
-  s.version     = '1.0.2'
+  s.version     = '1.0.3'
   s.executables << 'yi_appium_caps_util'
-  s.date        = '2017-02-10'
+  s.date        = '2017-10-04'
   s.summary     = "Updates your appium.txt capabilities for iOS and Android"
```
Comment (TamimiGitHub) [inline yi_appium_caps_util.gemspec]: I would keep this in a separate commit for documentations purposes

File: lib/yi_appium_caps_util.rb [hunk 2]
```diff
@@ -6,14 +6,14 @@
 class YiAppiumCapsUtil
   class << self
     public
-    def update (caps_file_name = './appium.txt')
+    def update (caps_file_name: './appium.txt', platformName_value: nil)
```
Comment (TamimiGitHub) [inline lib/yi_appium_caps_util.rb]: Why did you have to change `=` to `:`?

Comment (simongranger) [reply lib/yi_appium_caps_util.rb]: won't I be forced to populate caps_file_name first if I keep it with '='?

File: lib/yi_appium_caps_util.rb [hunk 3]
```diff
@@ -55,15 +55,16 @@ def getTeamID()
...
 
       if parsed_data['caps'] == nil
         raise '[caps] is missing form appium.txt'
       else
-        platformName_value = parsed_data['caps']['platformName']
-
+        if platformName_value == nil
+          platformName_value = parsed_data['caps']['platformName']
+        end
         if platformName_value == nil
```
Comment (TamimiGitHub) [inline lib/yi_appium_caps_util.rb]: i'm not sure if I got this correct, is `if platformName_value == nil` repeated?

Comment (simongranger) [reply lib/yi_appium_caps_util.rb]: if no `platformName_value` is passed, try to get it from the caps.
if you can get it from caps, then raise an exception

File: bin/yi_appium_caps_util [hunk 4]
```diff
@@ -29,8 +30,12 @@ end.parse!
 raise "You should pass -u or -c PLATFORM as an input argument. Type -h for help" if options[:update] == nil and options[:create_caps] == nil
 
 if options[:update]
-  if options[:file]
-    YiAppiumCapsUtil.update(options[:file])
+  if options[:file] && options[:platform]
+    YiAppiumCapsUtil.update(caps_file_name: options[:file], platformName_value: options[:platform])
```
Comment (TamimiGitHub) [inline bin/yi_appium_caps_util]: I dont think you would need all those ifelses, just `YiAppiumCapsUtil.update(caps_file_name: options[:file], platformName_value: options[:platform])` and in `update` you are already handling if the platformName is nill, right?

File: lib/yi_appium_caps_util.rb (hunk 2 above)
Comment (TamimiGitHub) [reply lib/yi_appium_caps_util.rb]: Then if thats the case you might not need all the ifelses in the `update` call, just call it once with the same parameters. I like the use `:`

File: lib/yi_appium_caps_util.rb (hunk 3 above)
Comment (TamimiGitHub) [reply lib/yi_appium_caps_util.rb]: Yeaa I see what we're trying to achieve here.. I just feel there is a neater way to do it in ruby by leveraging `unless` statements. So maybe something along the lines

```
      raise '[caps] is missing form appium.txt' unless parsed_data['caps'] != nil
      if platformName_value == nil then platformName_value = parsed_data['caps']['platformName']
      raise 'platformName is missing from appium.txt' unless platformName_value != nil

      case platformName_value.downcase
      when 'android'
        update_android_caps (output_data)
      when 'ios'
        update_ios_caps (output_data)
      else
        raise 'platformName: ' + platformName_value + ' is not supported'
      end

      return output_data
```

But other than that, logic looks good
